### ISSUE TITLE
Crawling does not work properly when using own deployment of prerender

### ISSUE THREAD

**Issue by technicalbirdVayuz** [prerender/prerender#774]
Hi,

We are using our own deployment of pre-render, i.e running prerender on port 4000, and updating the service URL as HTTP://website.com:4000/ 

if we request from the Google search console for a specific URL, logs also come but screenshot not come in the Google search console panel after the live test
<img width="1440" alt="Screenshot 2023-12-01 at 12 29 31 PM" src="https://github.com/prerender/prerender/assets/18699624/960af4c2-6319-4a1c-9c22-85f5c1128db6">

Whereas if we are using a prerender service URL i.e HTTP://service.prerender.io/ as the service URL then it works fine, Please explain what could be the issue

**Comment by technicalbirdVayuz** [comment]
Any update on this?

**Comment by turicas** [comment]
Are you proxying bot requests for your prerender instances (as you would for the prerenderi.io service)?
If no, please do so and test again. If yes, you must provide the logs and other details. Without it it's difficult to help.

**Comment by technicalbirdVayuz** [comment]
This is what I have done to setup
- Configured pre-render, cloned from https://github.com/prerender/prerender
- Running it at port 3100, while the application is running on port 3000 (reverse proxy is done at application through nginx)
- Now if I use command, curl http://localhost:3100/render?url=http://localhost:3000/feed/task/how-to-conv
ert-bsnl-3g-sim-to-4g-and-complete-the-activation-process (To render one of the page), It returns HTML but body does not have anything and that's why empty page is rendered at Google Console too. 

Doing the same with prerender.io as a service (with token) works

<img width="1094" alt="Screenshot 2024-02-28 at 1 12 07 PM" src="https://github.com/prerender/prerender/assets/18699624/6f7ecdba-1c80-4709-8268-935a94d8fddb">


<img width="1425" alt="Screenshot 2024-02-28 at 1 12 27 PM" src="https://github.com/prerender/prerender/assets/18699624/5b4fe042-8c27-46b9-bd87-b833d163881e">

Please help

**Comment by turicas** [comment]
Did you emailed any plugins? Which configuration (like env vars) you changed?
I had to enable some plugins and test some different configurations to have it working properly for a website that takes a while to load. Check the plugins enabled by default in https://github.com/PythonicCafe/dokku-prerender (`env.example` file).

**Comment by technicalbirdVayuz** [comment]
These plugins are not available with this in this code: https://github.com/prerender/prerender, can you check once @turicas

**Comment by turicas** [comment]
> These plugins are not available with this in this code: https://github.com/prerender/prerender, can you check once @turicas

Yes they are. https://github.com/prerender/prerender/tree/master/lib/plugins
Please read the documentation and try enabling them.

**Comment by technicalbirdVayuz** [comment]
Following are the plugins that are enabled but still page does not rendered as expected

```
server.use(prerender.sendPrerenderHeader());
server.use(prerender.browserForceRestart()); 
server.use(prerender.addMetaTags());
server.use(prerender.removeScriptTags());
server.use(prerender.httpHeaders());
```

**Comment by turicas** [comment]
Sorry, without more information on the page being rendered I can't help you.
